### PR TITLE
[clang-cl] [Driver] Fix clang-cl driver supported colon options

### PR DESCRIPTION
**Pull request by MaxEW707**
Tested locally against msvc 1939.
MSVC supports the colon form of various options that take a path even though that isn't documented for all options on MSDN.
I added the colon form for all supported msvc options that clang-cl does not intentionally ignore due to being unsupported.

I also fixed the existing colon options by ensure we use `JoinedOrSeparate`. `/F[x]` argument must used within the same command line argument. However, `/F[x]:` arguments are allowed to span a command line argument. The following is valid `/F[x]:     path` which is 2 separate command line arguments.
You can see this documented here, https://learn.microsoft.com/en-us/cpp/build/reference/fo-object-file-name?view=msvc-170, where it says `/Fo:[ ]"pathname"`.
This behaviour works for all colon options that take a path and I tested it locally against msvc 1939 for all the option changes in this PR.

### CODE REVIEW

**Comment by MaxEW707** [comment]
CC @rnk

**Review by rnk — APPROVED** [review]
Thank you for polishing this corner of the driver interface! It's interesting that they have an alternative separate spelling. I always felt like the `/Fopath.cpp` pattern was a bit unreadable.

**Comment by MaxEW707** [comment]
> Thank you for polishing this corner of the driver interface! It's interesting that they have an alternative separate spelling. I always felt like the /Fopath.cpp pattern was a bit unreadable.

Thanks for the review. I will need you to commit on my behalf. I think its about time I figure out what conditions I need to meet in-order to get commit after approval rights.

`/Fopath.cpp` is super unreadable. Unfortunately, I don't remember when the colon spelling came into existence but anything from VS2017 definitely has the colon alternative spelling.

**Pull request merged**
